test/utils.spec.js

```diff
@@ -22,12 +22,12 @@ describe('utils', () => {
 
   describe('#getPositionOfCapturingGroup', () => {
     const testData = {
-      '@[__display__](__id__)': { display: 0, id: 1, type: null },
-      '@[__display__](__type__:__id__)': { display: 0, id: 2, type: 1 },
-      '@(__type__:__id__)': { display: 1, id: 1, type: 0 },
-      '{{__id__#__display__}}': { display: 1, id: 0, type: null },
-      '{{__id__}}': { display: 0, id: 0, type: null },
-      '{{__display__}}': { display: 0, id: 0, type: null },
+      '@[__display__](__id__)': { display: 1, id: 2, type: null },
```

Comment:
The type of change for the spec makes it look like a breaking API change. Is that the case? I'm also missing a specific test case for mentions that use a `regex` pattern.

Reply:
Is `getPositionOfCapturingGroup` considered a public API ?
I could probably revert this by changing back the `getPlainText` method, but I guess it makes more sense like this. Again, I would be interested in @jfschwarz input since it is his original modifications (from the `custom-regex` branch)

Reply:
I'd actually need to check this. We do pass props that sound similar to the `render` methods that can be influenced by the user. That's what got me worried. @jfschwarz is on vacation for the next 4 weeks which is why he might not be able to chime in.

Reply:
Thanks for your answer :) And enjoy your vacation, @jfschwarz ! Anyway this is not crucial to this PR, so I just reverted the changes, and confirmed that it was still working for me.

The method _is_ exported utils, but _this comment_ tells me it was probably not meant to be publicly available : 
```javascript
/**
 * parameterName: "id", "display", or "type"
 * TODO: This is currently only exported for testing
 */
export const getPositionOfCapturingGroup = (markup, parameterName, regex) => {
```